users/engler/fairseq/training.py

```diff
@@ -0,0 +1,162 @@
+        :param Path model: Path to a pytorch checkpoint
+        :param int epoch:
+        """
+        self.returnn_config_file = fairseq_hydra_config_file
+        self.model = model
+        self.epoch = epoch
+
+
+class FairseqHydraTrainingJob(Job):
+    """
+    Train a Fairseq model using fairseq-hydra-train
+    """
+
+    def __init__(
+        self,
+        fairseq_hydra_config,
+        *,  # args below are keyword only
+        command_line_args=None,
+        max_epoch=1,
+        save_interval=1,
+        keep_epochs=None,
+        time_rqmt=4,
+        mem_rqmt=4,
+        cpu_rqmt=2,
+        gpu_rqmt=1,
+        fairseq_python_exe=None,
+        fairseq_hydra_exe=None,
+        fairseq_git=None,
+    ):
+        """
+        :param FairseqHydraConfig fairseq_hydra_config:
+        :param list command_line_args: Additional command line arguments (starting with "--*") to configure the Fairseq-hydra task
+        :param int max_epoch: maximum number of epochs to run. Note that this value IS currently HASHED.
+        :param int save_interval: save a checkpoint each n-th epoch
+        :param list[int]|set[int]|None keep_epochs: specify which checkpoints are kept in self.out_models, use None for each save_interval-th epoch
+        :param int|float time_rqmt: Overall time requirements
+        :param int|float mem_rqmt: Memory requirements (per GPU)
+        :param int cpu_rqmt: Required number of CPUs (per GPU)
+        :param int gpu_rqmt: Number of required GPUs
+        :param Path|str python_exe: File path to the executable for running python
+        :param Path|str fairseq_hydra_exe: File path to the Fairseq executable: fairseq-hydra-train (usually in the same folder as python exe '.../bin/')
+        :param Path|str fairseq_git: File path to the fairseq git for alternative call of fairseq-hydra-train (no need to install fairseq here)
+
+        """
+        self.fairseq_hydra_config = fairseq_hydra_config
+        self.command_line_args = command_line_args or []
+
+        stored_epochs = list(range(save_interval, max_epoch, save_interval)) + [
+            max_epoch
+        ]
+        if keep_epochs is None:
+            self.keep_epochs = set(stored_epochs)
+        else:
+            self.keep_epochs = set(keep_epochs)
+        self.max_epoch = max_epoch
+        self.save_interval = save_interval
+        self.out_fairseq_hydra_yaml = self.output_path("fairseq_hydra_config.yaml")
+        self.out_checkpoint_dir = self.output_path("checkpoints", directory=True)
+        self.out_models = {
+            k: PytorchHydraModel(
+                self.out_fairseq_hydra_yaml,
+                self.output_path("checkpoints/checkpoint{}.pt".format(k)),
+                k,
+            )
+            for k in stored_epochs
+            if k in self.keep_epochs
+        }
+        self.gpu_rqmt = gpu_rqmt
+        self.fairseq_python_exe = fairseq_python_exe
```

Comment:
Maybe also allow to set a default in the global settings
```suggestion
        self.fairseq_python_exe = (
            fairseq_python_exe if fairseq_python_exe is not None else gs.FAIRSEQ_PYTHON_EXE
        )
```